kolibri/plugins/user/middleware.py

```diff
@@ -0,0 +1,36 @@
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.shortcuts import redirect
+from django.utils.deprecation import MiddlewareMixin
+
+from kolibri.core.auth.models import Facility
+from kolibri.core.auth.models import KolibriAnonymousUser
+
+ALLOWED_PATH_LIST = [
+    'deviceprovision',
+    'kolibri:setupwizardplugin:setupwizard',
+    'kolibri:set_language',
+    'session-list'
+    'admin:index',
+]
+
+
+class RedirectToSignInPageIfNoGuestAccessAndNoActiveSession(MiddlewareMixin):
```

Comment:
Will something like this https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/setup_wizard/middleware.py#L17 be needed for this middleware as well? cc: @rtibbles

Reply:
Fortunately, this change is now superfluous thanks to: https://github.com/learningequality/kolibri/blob/develop/kolibri/core/device/middleware.py#L16 - this shortcuts any other middleware in the case that the user agent is the Windows GUI.